#CENARIO DA ANALISE

-ARQUIVO: titanic.csv

DICIONÁRIO DE DADOS

-Survived (Sobreviveu): 
0 = Não, 1 = Sim

-Pclass (Classe): Classe de ingresso 
1 = 1º, 2 = 2º, 3 = 3º

Sex (Sexo): Sexo do passageiro


Age (Idade): Idade em anos


Sibsp: Quantidade de irmãos / cônjuges a bordo do Titanic
Parch: Quantidade de pais / crianças a bordo do Titanic
Ticket (Bilhete): Número do bilhete de embarque
Fare (Tarifa): Tarifa paga pelo Passageiro
Cabin (Cabine): Número de cabine
Embarked (Embarque): Porto de Embarque (C = Cherbourg, Q = Queenstown, S = Southampton)

Notas sobre as variáveis

Pclass (Classe): 1º = Superior 2º = Médio 3º = inferior

Age (Idade): A idade é fracionada se for inferior a 1. Se a idade for estimada, é na forma de xx.5

Sibsp: O conjunto de dados define as relações familiares dessa maneira.


Sibling = Irmão, irmã, meio-irmão, irmandade
Spouse (Cônjuge) = marido, esposa (amantes e desposados foram ignorados)



Objetivo da analisando o registro de bordo do Titanic e extrair informações gerais e especificas

-Quantidade total de passageiros e sobreviventes, 
-MEDIA DE IDADE POR SEXO


CARREGAMENTO DO ARQUIVO


In [3]:
import pandas as pd

df = pd.read_csv("C:/Users/Papai/Google Drive/5.Projetos/estudopy/estudopy/arquivos/titanic.csv")

SyntaxError: invalid syntax (3943192108.py, line 1)

In [38]:
display(df.info())
display(df.isnull().sum())
display(df.describe())
display(df)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


None

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [39]:
df = df.drop(["PassengerId"], axis =1)
df = df.drop(["SibSp"], axis =1)
df = df.drop(["Parch"], axis =1)
df = df.drop(["Ticket"], axis =1)
df = df.drop(["Cabin"], axis =1)

In [40]:
display(df.info())
display(df.isnull().sum())
display(df.describe())
display(df)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Name      891 non-null    object 
 3   Sex       891 non-null    object 
 4   Age       714 non-null    float64
 5   Fare      891 non-null    float64
 6   Embarked  889 non-null    object 
dtypes: float64(2), int64(2), object(3)
memory usage: 48.9+ KB


None

Survived      0
Pclass        0
Name          0
Sex           0
Age         177
Fare          0
Embarked      2
dtype: int64

Survived      Pclass         Age        Fare
count  891.000000  891.000000  714.000000  891.000000
mean     0.383838    2.308642   29.699118   32.204208
std      0.486592    0.836071   14.526497   49.693429
min      0.000000    1.000000    0.420000    0.000000
25%      0.000000    2.000000   20.125000    7.910400
50%      0.000000    3.000000   28.000000   14.454200
75%      1.000000    3.000000   38.000000   31.000000
max      1.000000    3.000000   80.000000  512.329200

Survived  Pclass                                               Name  \
0           0       3                            Braund, Mr. Owen Harris   
1           1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2           1       3                             Heikkinen, Miss. Laina   
3           1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4           0       3                           Allen, Mr. William Henry   
..        ...     ...                                                ...   
886         0       2                              Montvila, Rev. Juozas   
887         1       1                       Graham, Miss. Margaret Edith   
888         0       3           Johnston, Miss. Catherine Helen "Carrie"   
889         1       1                              Behr, Mr. Karl Howell   
890         0       3                                Dooley, Mr. Patrick   

        Sex   Age     Fare Embarked  
0      male  22.0   7.2500        S  
1    female  38.0  71.2833        C  
2    female  26.0   7.9250        S  
3    female  35.0  53.1000        S  
4      male  35.0   8.0500        S  
..      ...   ...      ...      ...  
886    male  27.0  13.0000        S  
887  female  19.0  30.0000        S  
888  female   NaN  23.4500        S  
889    male  26.0  30.0000        C  
890    male  32.0   7.7500        Q  

[891 rows x 7 columns]

1- INICIANDO A TRATAMENTO DO DADOS
-TRATAMENTO DE CAMPOS VAZIO 
COLUNAS COM REGISTRO VAZIOS OU NULOS

COLUNA    TOTAL VAZIOS OU NULOS
Age         177
Embarked      2

-RESOLVENDO O PROBLEMA DA COLUNA Embarked COM DOIS REGISTROS VAZIOS E NÃO NUMERICOS
VERIFICO AS REPETIÇÕES DO REGISTRO

In [54]:
df["Embarked"].value_counts(normalize=False)

S    644
C    168
Q     77
Name: Embarked, dtype: int64

In [59]:
embarked_MaiorFreq = "S"

In [60]:
df["Embarked"].fillna(embarked_MaiorFreq , inplace=True)

In [61]:
df["Embarked"].value_counts()

S    646
C    168
Q     77
Name: Embarked, dtype: int64

In [64]:
df["Embarked"].mode()

0    S
Name: Embarked, dtype: object

-RESOLVENDO O PROBLEMA DA COLUNA Age COM 177 REGISTROS VAZIOS E NUMERICOS 
-Verificando a moda, media e mediana

In [79]:
df.mode(axis =0 , numeric_only = True , dropna = True)

Survived  Pclass   Age  Fare
0         0       3  24.0  8.05

In [88]:
df.mean(axis =0, skipna=True )

C:\Users\Papai\AppData\Local\Temp\ipykernel_9188\1699688053.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df.mean(axis =0, skipna=True )


Survived     0.383838
Pclass       2.308642
Age         29.699118
Fare        32.204208
dtype: float64

In [89]:
df["Age"].mean(axis =0, skipna=True )

29.69911764705882

In [86]:
df.median(axis =0, skipna=True)

C:\Users\Papai\AppData\Local\Temp\ipykernel_9188\523233236.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df.median(axis =0, skipna=True)


Survived     0.0000
Pclass       3.0000
Age         28.0000
Fare        14.4542
dtype: float64

REMOVENDO A COLUNA COM INFORMAÇÕES DESNECESSARIAS 
axis = 0 ou "index" será apagada a linha indicada
axis = 1 ou  "columns" será apagada a coluna indicada

In [93]:
age_Mediana = df["Age"].median(axis =0, skipna=True)

In [94]:
df["Age"].fillna(age_Mediana , inplace=True)

In [95]:
display(df.info())
display(df.isnull().sum())
display(df.describe())
display(df)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Name      891 non-null    object 
 3   Sex       891 non-null    object 
 4   Age       891 non-null    float64
 5   Fare      891 non-null    float64
 6   Embarked  891 non-null    object 
dtypes: float64(2), int64(2), object(3)
memory usage: 48.9+ KB


None

Survived    0
Pclass      0
Name        0
Sex         0
Age         0
Fare        0
Embarked    0
dtype: int64

Survived      Pclass         Age        Fare
count  891.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.361582   32.204208
std      0.486592    0.836071   13.019697   49.693429
min      0.000000    1.000000    0.420000    0.000000
25%      0.000000    2.000000   22.000000    7.910400
50%      0.000000    3.000000   28.000000   14.454200
75%      1.000000    3.000000   35.000000   31.000000
max      1.000000    3.000000   80.000000  512.329200

Survived  Pclass                                               Name  \
0           0       3                            Braund, Mr. Owen Harris   
1           1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2           1       3                             Heikkinen, Miss. Laina   
3           1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4           0       3                           Allen, Mr. William Henry   
..        ...     ...                                                ...   
886         0       2                              Montvila, Rev. Juozas   
887         1       1                       Graham, Miss. Margaret Edith   
888         0       3           Johnston, Miss. Catherine Helen "Carrie"   
889         1       1                              Behr, Mr. Karl Howell   
890         0       3                                Dooley, Mr. Patrick   

        Sex   Age     Fare Embarked  
0      male  22.0   7.2500        S  
1    female  38.0  71.2833        C  
2    female  26.0   7.9250        S  
3    female  35.0  53.1000        S  
4      male  35.0   8.0500        S  
..      ...   ...      ...      ...  
886    male  27.0  13.0000        S  
887  female  19.0  30.0000        S  
888  female  28.0  23.4500        S  
889    male  26.0  30.0000        C  
890    male  32.0   7.7500        Q  

[891 rows x 7 columns]

FIM DAS ETAPAS DE EXTRAÇÃO E TRATAMENTO DOS DADOS

3- INICIANDO ANALISE EXPLORATORIA / APRESENTAÇÃO DOS GRAFICOS

Objetivo da analisando o registro de bordo do Titanic e extrair informações gerais e especificas

-Quantidade total de passageiros e sobreviventes, 
-MEDIA DE IDADE POR SEXO

#Analise exploratória
-PRINCIPAL DESTINO,
-QUANTIDADE DE PASSAGEIROS E SOBREVIVENTES POR SEXO
-MEDIA DE IDADE POR SEXO
-MEDIA DE IDADE

#retorna qtd em inteiro percentual
display(df["Churn"].value_counts())

#retorna qtd em percentual
display(df["Churn"].value_counts(normalize="True"))
#retorno com uma casa decimal
.map("{:.1%}".format)
display(df["Churn"].value_counts(normalize="True").map("{:.1%}".format) )


Iniciando a Análise gráfica
importando ploty
import plotly.express as px

In [99]:
import plotly.express as px

# para edições nos gráficos: https://plotly.com/python/histograms/

for coluna in df:
    if coluna != "Survived":
        # criar a figura
        fig = px.histogram(df, x=coluna, color="Sex")
        # exibir a figura
        fig.show()
             